### PR TITLE
Fixit

### PR DESCRIPTION
**Pull request by RogerHaase**

### CODE REVIEW

File: src/moin/storage/middleware/protecting.py [hunk 1]
```diff
@@ -340,6 +340,14 @@ def __init__(self, protector, rev, p_item=None):
         self.rev = rev
         self.item = p_item or ProtectedItem(protector, rev.item)
 
+    def __str__(self):
+        """
+        Return revision contents as a string.
+        """
+        self.data.seek(0)
```
Comment (ThomasWaldmann) [inline src/moin/storage/middleware/protecting.py]: that looks strange. where does self.data come from?

did you mean `self.rev.data`?

also, this does not really belong into the protecting middleware, so i guess here there should be only a stub that delegates to `rev.__str__` and the implementation should be there.

Comment (ThomasWaldmann) [reply src/moin/storage/middleware/protecting.py]: also, as this is just another way to get the data, guess we need the same permission checks as in the `data` property, see below.

File: src/moin/scripts/maint/modify_item.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ def run(self, name, meta_file, data_file, revid):
         if 'charset' in rev.meta['contenttype']:
             # input data will have \r\n line endings, output will have platform dependent line endings
             charset = rev.meta['contenttype'].split('charset=')[1]
-            data = rev.data.read().decode(charset)
+            data = str(rev.data)
```
Comment (ThomasWaldmann) [inline src/moin/scripts/maint/modify_item.py]: did you want to call `str(rev)`?

Comment (RogerHaase) [reply src/moin/scripts/maint/modify_item.py]: No that rev is not a ProtectedRevision, so it must be reverted to the old code:

    data = rev.data.read().decode(charset)

File: src/moin/storage/middleware/protecting.py (hunk 1 above)
Comment (RogerHaase) [reply src/moin/storage/middleware/protecting.py]: self is a ProtectedRevision object. I think it is best to revert 4024102 and 708ec5d and reopen #890. 

I do not understand what you meant by best = "text = str(rev): in the #890 issue.

Also, seems there is very little code saved.

File: src/moin/scripts/maint/modify_item.py (hunk 2 above)
Comment (ThomasWaldmann) [reply src/moin/scripts/maint/modify_item.py]: no, the code implementing `__str__` must be in the Revision, the ProtectedRevision then just forwards to that.

File: src/moin/storage/middleware/protecting.py (hunk 1 above)
Comment (ThomasWaldmann) [reply src/moin/storage/middleware/protecting.py]: I meant implementing `__str__`.

Just do not put code that has nothing to do with "protection" (== permissions/acl checking) into the protection middleware.

That functionality is rather fundamental and thus should be implemented in the Revision class.
And of course there also needs to be a __str__ in the ProtectedRevision, but that would just check permisssions and then forward to the unprotected Revision.

Comment (RogerHaase) [reply src/moin/storage/middleware/protecting.py]: ProtectedRevision already has:

    @property
    def meta(self):
        self.require(READ, PUBREAD)
        return self.rev.meta

    @property
    def data(self):
        self.require(READ, PUBREAD)
        return self.rev.data

The above seem to get called when .data is referenced by a call to the `__str__`  method, so no need to check again.

Comment (ThomasWaldmann) [reply src/moin/storage/middleware/protecting.py]: You would not call these, but `str(self.rev)` - and before doing that, check permissions (only once).

**Pull request merged**